### PR TITLE
PatternsHW

### PR DESCRIPTION
**Pull request by Dinexpod**

### CODE REVIEW

File: src/main/java/mate/academy/HWpatterns/Adapter/AdapterToSmartphone.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mate.academy.HWpatterns.Adapter;
+
+public class AdapterToSmartphone extends Smartphone implements Phone {
+    Smartphone myPhone = new Smartphone();
+
+    @Override
+    public void pushToCall() {
+        myPhone.touchToCall();
```
Comment (boroda4436) [inline src/main/java/mate/academy/HWpatterns/Adapter/AdapterToSmartphone.java]: +1 Adapter

File: src/main/java/mate/academy/HWpatterns/flyweightAndPrototype/DevFactory.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class DevFactory {
+    private static final Map<String, Developer> developers = new HashMap<>();
+
+    public Developer getDevSpeciality(String speciality) {
+        Developer developer = developers.get(speciality);
+
+        if (developer == null) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/HWpatterns/flyweightAndPrototype/DevFactory.java]: +1 flyweight

File: src/main/java/mate/academy/patternsOnLesson/BuildPatternExample.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.patternsOnLesson;
+
+public class BuildPatternExample implements PatternExample {
+    public Pattern makePattern() {
+        return BuilderPattern.builder().build();
```
Comment (boroda4436) [inline src/main/java/mate/academy/patternsOnLesson/BuildPatternExample.java]: +1 Builder

File: src/main/java/mate/academy/patternsOnLesson/BuilderPattern.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package mate.academy.patternsOnLesson;
+
+public class BuilderPattern implements Pattern {
+    private String name;
+    private int age;
+
+    public BuilderPattern(String name, int age) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/patternsOnLesson/BuilderPattern.java]: Should this one need to be public?

File: src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package mate.academy.patternsOnLesson;
+
+public enum SingletonPattern implements Pattern {
+    INSTANCE;
```
Comment (boroda4436) [inline src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java]: Where is INSTANCE initializing?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/patternsOnLesson/BuilderPattern.java (hunk 4 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/patternsOnLesson/BuilderPattern.java]: Maybe no) Inner class use this constructor, so this one can be "privat"(I changed to privat)

File: src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java (hunk 5 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java]: image
here) or I don't understand.

Comment (boroda4436) [reply src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java]: INSTANCE on the picture is just an example. You need to initialize it. What is the data type of your INSTANCE? Is it String, Object or maybe some your custom class like User, Car ....

Comment (Dinexpod) [reply src/main/java/mate/academy/patternsOnLesson/SingletonPattern.java]: image
image

**Pull request merged**